cli/connhelper/connhelper.go

```diff
@@ -102,21 +102,30 @@ func (c *commandConn) killIfStdioClosed() error {
 	if !stdioClosed {
 		return nil
 	}
-	var err error
 	c.cmdMutex.Lock()
-	// NOTE: maybe already killed here
-	if err = c.cmd.Process.Kill(); err == nil {
-		err = c.cmd.Wait()
+	killErr := c.cmd.Process.Kill()
+	waitErr := c.cmd.Wait()
+	// NOTE: checking ProcessState before Wait() result in nil panic
+	exited := c.cmd.ProcessState.Exited()
```

Comment:
Can we read process state from `waitErr` instead? (e.g. `waitErr.(*exec.ExitError)`)
I'm not a big fan of the error handling here, and seems pretty platform dependent.
It also seems like we'd run into  the `signal: killed` error in the normal case?

Should we also consider sending TERM instead of KILL and only send kill if TERM doesn't exit after some time?

Reply:
addressed the comment, PTAL